pkg/aws/metadata/handler_credentials.go

```diff
@@ -90,15 +66,14 @@ func (c *credentialsHandler) Handle(ctx context.Context, w http.ResponseWriter,
 	return http.StatusOK, nil
 }
 
-func (c *credentialsHandler) credentialsForRole(ctx context.Context, role string) (*sts.Credentials, error) {
+func (c *credentialsHandler) fetchCredentials(ctx context.Context, ip, requestedRole string) (*sts.Credentials, error) {
```

Comment:
this is a great improvement and reduces the ability of compromised client to just arbitrarily request roles.